src/ser-ext-ondemandDirective.ts

```diff
@@ -380,8 +382,14 @@ class OnDemandController implements ng.IController {
         if (typeof (intervalTime) === "undefined") {
             intervalTime = 5000;
         }
-        this.interval = window.setInterval(() => {
-            this.getStatus(this.taskId);
+        this.interval = window.setInterval(async () => {
+            this.timeoutResponseCounter++;
+            if (this.timeoutResponseRevieved || this.timeoutResponseCounter > 10) {
+                this.timeoutResponseRevieved = false;
```

Comment:
should be redundant since u set this to false and 1 line later to response from promise, could removed

Reply:
@r-hannuschka : set false at this point is necessary, because when the response from the getStatus Promise takes a lot of time, it will not enter the getStatus request again